features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandlerTest.kt

```diff
@@ -393,6 +413,38 @@ internal class RecordedDataQueueHandlerTest {
         verifyNoMoreInteractions(mockProcessor)
     }
 
+    @Test
+    fun `M remove item from queue W tryToConsumeItems() { invalid resource item }`() {
+        // Given
+        val spy = spy(fakeResourceItem)
+
+        doReturn(false).whenever(spy).isValid()
```

Comment:
can we just create `fakeResourceItem` which will have `isValid` evaluated to `false` instead of involving `spy`? not a big deal, but usually `spy` is kind of hack.